src/world/area_end/end_00/DF9300.c

```diff
@@ -1,7 +1,18 @@
 #include "end_00.h"
-
+#include "battle/battle.h"
 #include "world/common/PlayFX3D.inc.c"
 
 INCLUDE_ASM(s32, "world/area_end/end_00/DF9300", func_80242A30_DF9430);
 
-INCLUDE_ASM(s32, "world/area_end/end_00/DF9300", func_80242AD4_DF94D4);
+typedef struct unk_DF94D4 {
+    /* 0x00 */ char unk00[0x04];
+    /* 0x04 */ s32 unk04;
+} unk_DF94D4;
+
+ApiStatus func_80242AD4_DF94D4(Evt *script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    EffectInstance* effect = (EffectInstance*) evt_get_variable(script, *args++);
+
+    ((unk_DF94D4*) effect->data)->unk04 = 1;
```

Comment:
No need for the struct here imo if we're just matching it outright. I'd just cast to an s32* and use [1] - please also leave a comment saying "TODO use correct effect data struct" as well, since that's what this is supposed to be